python/core/qgsstringstatisticalsummary.sip

```diff
@@ -1,165 +1,220 @@
-     * @note call reset() before adding the first string using this method
-     * to clear the results from any previous calculations
-     * @note finalize() must be called after adding the final value and before
-     * retrieving calculated statistics.
-     * @see calculateFromVariants()
-     * @see finalize()
-     */
-    void addValue( const QVariant& value );
-
-    /** Must be called after adding all strings with addString() and before retrieving
-     * any calculated string statistics.
-     * @see addString()
-     */
     void finalize();
+%Docstring
+ Must be called after adding all strings with addString() and before retrieving
+ any calculated string statistics.
+.. seealso:: addString()
+%End
 
-    /** Returns the value of a specified statistic
-     * @param stat statistic to return
-     * @returns calculated value of statistic
-     */
     QVariant statistic( Statistic stat ) const;
+%Docstring
+ Returns the value of a specified statistic
+ \param stat statistic to return
+ :return: calculated value of statistic
+ :rtype: QVariant
+%End
 
-    /** Returns the calculated count of values.
-     */
     int count() const;
+%Docstring
+ Returns the calculated count of values.
+ :rtype: int
+%End
 
-    /** Returns the number of distinct string values.
-     * @see distinctValues()
-     */
     int countDistinct() const;
+%Docstring
+ Returns the number of distinct string values.
+.. seealso:: distinctValues()
+ :rtype: int
+%End
 
-    /** Returns the set of distinct string values.
-     * @see countDistinct()
-     */
     QSet< QString > distinctValues() const;
+%Docstring
+ Returns the set of distinct string values.
+.. seealso:: countDistinct()
+ :rtype: set of str
+%End
 
-    /** Returns the number of missing (null) string values.
-     */
     int countMissing() const;
+%Docstring
+ Returns the number of missing (null) string values.
+ :rtype: int
+%End
 
-    /** Returns the minimum (non-null) string value.
-     */
     QString min() const;
+%Docstring
+ Returns the minimum (non-null) string value.
+ :rtype: str
+%End
 
-    /** Returns the maximum (non-null) string value.
-     */
     QString max() const;
+%Docstring
+ Returns the maximum (non-null) string value.
+ :rtype: str
+%End
 
-    /** Returns the minimum length of strings.
-     */
     int minLength() const;
+%Docstring
+ Returns the minimum length of strings.
+ :rtype: int
+%End
 
-    /** Returns the maximum length of strings.
-     */
     int maxLength() const;
+%Docstring
+ Returns the maximum length of strings.
+ :rtype: int
+%End
 
-    /**
-      * Returns the mean length of strings.
-      * @note added in QGIS 3.0
-      */
     double meanLength() const;
+%Docstring
+ Returns the mean length of strings.
+.. versionadded:: 3.0
+ :rtype: float
+%End
 
-    /** Returns the friendly display name for a statistic
-     * @param statistic statistic to return name for
-     */
     static QString displayName( Statistic statistic );
```

Comment:
you might need to add `QgsStatisticalSummary`in front of `Statistic` (in the header)